### ISSUE TITLE
[DOC] documentation & developer instructions (Windows experience) fixing umbrella issue

### ISSUE THREAD

**Issue by fkiraly-shell** [alan-turing-institute/sktime#1635]
I've been trying to set up a sktime developer environment in my day job role, we have a standard hardware set-up that uses Windows laptops.
Unlike the unix/mac experience which seems smooth, I truly must say that the Windows experience of the developer instructions is quite a bit atrocious, even mere installation can be confusing.
We've done user testing with 5 people, and no one managed to complete the `sktime` developer set-up without any hassle.

So, we've decided to help by fixing this, with @eenticott-shell, @amrith-shell, by:

- [x] updating and validating the installation instructions in windows. Related issue: #1372, #1203
- [ ] completing and re-organizing the developer guide (currently mostly empty!) and the "how to contribute" page. Related issue: #1448, #1203

**Comment by amrith-shell** [comment]
I will pick up the 2nd task -  completing and re-organizing the developer guide (currently mostly empty!) and the "how to contribute" page.

**Comment by fkiraly** [comment]
Great (as discussed) - @amrith-shell, it would be nice if you could kindly outline what in your opinion should go where, i.e., a sketch of end state. Happy to help with that tomorrow, although I assume you might already have better ideas since you worked through this in the user journey.

**Comment by TonyBagnall** [comment]
@jasonlines has a version of this too, good to compare notes

**Comment by jasonlines** [comment]
yep, went through the pain of getting it up and running on windows last week so uninstalled everything to do it again yesterday from first principles and document it. I've taken screenshots throughout and the plan is to put it together in a doc/guide - happy to put it together as a notebook or similar initially and go from there?

**Comment by fkiraly** [comment]
great! Let's do this together then

**Comment by eenticott-shell** [comment]
I have some instructions written up in markdown that I was able to use for a successful install on a new windows pc. It will be interesting to see if we have the same path.

**Comment by jasonlines** [comment]
ok great, sounds good. I'll aim to get mine into a markdown by tomorrow lunchtime (hopefully the close of play today though) and we can compare notes before going from there

**Comment by jasonlines** [comment]
It took longer than planned but please see #1640 for my first pass at this

**Comment by amrith-shell** [comment]
In terms of the 2nd task of re-organizing the documentation/developer guide, I took the approach of categorizing various sections of the documentation and created a hierarchy/ decision tree of the documentation structure. Please see my initial draft.

If you have any comments or suggestions please do let  me know.

![documentation design](https://user-images.githubusercontent.com/83223497/143680784-ca0e95dc-b9d6-4520-b2bd-60cc7abbccd8.png)

**Comment by fkiraly** [comment]
Thanks!I think idea-wise this makes a lot of sense, excellent!

Few comments:
* you forgot "get started", that's perhaps the most important entry point! This should also link to "installation" directly. The "installation" page, in my opinion, should contain workflows for release install, development version install, and developer/build install (these are very similar).
* moving stuff from "contributing" to the developer guide makes sense. However I'd also link directly from "contributing" to the "developer guide", and possibly individual sub-sections.
* I think "code of conduct" should appear primarily under "get involved", and linked in "contributing" too. Any interaction with the project should be subject to the CoC.
* would "governance" make more sense under "get involved"?

PS: with my day job hat on (i.e., @fkiraly-shell), the project item scope is ensuring that the Windows install related documentation is unambiguous. I.e., this can imply moving some material around, but we'd focus on getting the install instructions "de-duplicated" and the structure of contributing/development cleaned up primarily.

**Comment by amrith-shell** [comment]
Updated documentation structure based on feedback 
![documentation design](https://user-images.githubusercontent.com/83223497/144220608-e59d41af-2072-4852-909b-d6807f87d229.png)

**Comment by eenticott-shell** [comment]
[Developer install sktime instructions.md](https://github.com/alan-turing-institute/sktime/files/7633142/Developer.install.sktime.instructions.md)

Here are my instructions written in markdown for windows developer installation. These should take you from start to finish (tested on my home computer). It remains to integrate them into the current installation instructions. I will set up a PR for updating the visual studio documentation and properly setting up a conda environment. 

Once the restructuring has been implemented, a small rework of the installation instructions could be done. In particular, I have realised a lot of the information that I needed to fix the problems I had was on the page, but it is in different places. (e.g. fbprophet fix)
